ibis/sql/tests/test_compiler.py

```diff
@@ -2335,3 +2335,71 @@ def test_having_from_filter():
 GROUP BY 1
 HAVING max(`a`) = 2"""
     assert result == expected
+
+
+def test_simple_agg_filter():
+    t = ibis.table([('a', 'int64'), ('b', 'string')], name='my_table')
+    filt = t[t.a < 100]
+    expr = filt[filt.a == filt.a.max()]
+    result = to_sql(expr)
+    expected = """\
+SELECT *
+FROM (
+  SELECT *
+  FROM my_table
+  WHERE `a` < 100
+) t0
+WHERE `a` = (
+  SELECT max(`a`) AS `max`
+  FROM my_table
+  WHERE `a` < 100
```

Comment:
Query looks good, but based on the SQL I'd check that it works correctly with aliasing:

```py
expr = filt[filt.a.name('A') == filt.a.max()]
```